packages/core/src/test/docdb/utils.test.ts

```diff
@@ -0,0 +1,138 @@
+        const message = validateUsername('x'.repeat(64))
+        assert.strictEqual(message, 'Username name must be between 1 and 63 characters long')
+    })
+
+    it('Validates username starts with a letter', function () {
+        const message = validateUsername('123')
+        assert.strictEqual(message, 'Username must start with a letter')
+    })
+
+    it('Allows usernames with letters and numbers', function () {
+        const message = validateUsername('a2Z')
+        assert.strictEqual(message, undefined)
+    })
+})
+
+describe('validatePassword', function () {
+    it('Validates password is not too short', function () {
+        const message = validatePassword('1234567')
+        assert.strictEqual(message, 'Password must be between 8 and 100 characters long')
+    })
+
+    it('Validates password is not too long', function () {
+        const message = validatePassword('x'.repeat(101))
+        assert.strictEqual(message, 'Password must be between 8 and 100 characters long')
+    })
+
+    it('Validates password does not include slash, double quote or @ symbol', function () {
+        ;['pass/word', 'p@ssword', '"password"'].forEach(item =>
+            assert.strictEqual(
+                validatePassword(item),
+                'Password must only contain printable ASCII characters (except for slash, double quotes and @ symbol)'
+            )
+        )
+    })
+
+    it('Validates password does not include non-printable ASCII characters', function () {
+        ;['password\x19', 'password\u{fe0f}'].forEach(item =>
+            assert.strictEqual(
+                validatePassword(item),
+                'Password must only contain printable ASCII characters (except for slash, double quotes and @ symbol)'
+            )
+        )
+    })
+
+    it('Allows passwords with printable ASCII characters', function () {
+        const message = validatePassword('passw0rd |~')
+        assert.strictEqual(message, undefined)
+    })
+})
+
+describe('validateInstanceName', function () {
+    it('Validates instance name is not blank', function () {
+        const message = validateInstanceName('')
+        assert.strictEqual(message, 'Instance name must be between 1 and 63 characters long')
+    })
+
+    it('Validates instance name is not too long', function () {
+        const message = validateInstanceName('c'.repeat(64))
+        assert.strictEqual(message, 'Instance name must be between 1 and 63 characters long')
+    })
+
+    it('Validates instance name starts with a lowercase letter', function () {
+        const message = validateInstanceName('404')
+        assert.strictEqual(message, 'Instance name must start with a lowercase letter')
+    })
+
+    it('Validates instance name does not contain uppercase characters', function () {
+        const message = validateInstanceName('abcDEF')
+        assert.strictEqual(message, 'Instance name must only contain lowercase letters, numbers, and hyphens')
+    })
```

Comment:
there is no reason for these tests to have separate it() cases, especially when the assert() has an explanatory message.

please look for ways to be less verbose in all parts of this PR.